fairing/notebook/jobs.py

```diff
@@ -0,0 +1,61 @@
+import fairing
+from fairing.functions.function_shim import get_execution_obj_type, ObjectType
+from fairing.preprocessors.function import FunctionPreProcessor
+from fairing.builders.docker.docker import DockerBuilder
+from fairing.deployers.job.job import Job
+from fairing.deployers.serving.serving import Serving
+
+def guess_preprocessor(obj, *args, **kwargs):
+    if get_execution_obj_type(obj) != ObjectType.NOT_SUPPORTED:
+        return FunctionPreProcessor(obj, *args, **kwargs) 
+    else:
+        raise NotImplementedError("obj param should be a function or a class, got {}".format(type(obj)))
+
+def guess_docker_registry():
```

Comment:
these should be handled lower down the stack, possibly in the base builder

Reply:
Removed guess_docker_registry and made docker registry required param for now. Let's address how to provide good defaults based on the environment (KF on-prem, KF on GKE, managed backend, etc)